### PR TITLE
Enable CompileBindings everywhere globally in the solution

### PR DESCRIPTION
**Pull request by maxkatz6**
## What is the current behavior?

ReflectionBinding is used in some places.

## What is the updated/expected behavior with this PR?

CompiledBinding is used everywhere where possible. And is enabled globally.

### CODE REVIEW

File: src/Avalonia.Dialogs/ManagedFileChooserFilterViewModel.cs [hunk 1]
```diff
@@ -1,10 +1,11 @@
 using System.Linq;
 using System.Text.RegularExpressions;
+using Avalonia.Dialogs.Internal;
 using Avalonia.Platform.Storage;
 
 namespace Avalonia.Dialogs
```
Comment (kekekeks) [inline src/Avalonia.Dialogs/ManagedFileChooserFilterViewModel.cs:6]: Inconsistency: `InternalViewModelBase` got moved to `Internal` namespace. I think the same should happen to those view models.

Comment (maxkatz6) [reply src/Avalonia.Dialogs/ManagedFileChooserFilterViewModel.cs:6]: Yes, I can do this.
My point was to make "InternalViewModelBase" less visible, as it should be used by anybody. While other dialog view models can be used in control restyling. But it also makes sense to move them to this namespace all anyway, as it's not common to change these styles.

File: src/Markup/Avalonia.Markup.Xaml.Loader/CompilerExtensions/Transformers/AvaloniaXamlIlSelectorTransformer.cs [hunk 2]
```diff
@@ -208,17 +208,9 @@ private static IXamlType GetLastTemplateTypeFromSelector(XamlIlSelectorNode node
...
-                    if (templateSelector.Previous is XamlIlNestingSelector nestingSelector)
-                    {
-                        return nestingSelector.TargetType;
-                    }
-                    if (templateSelector.Previous is XamlIlTypeSelector typeSelector)
-                    {
-                        return typeSelector.TargetType;
-                    }
-                    return null;
+                    return node.Previous.TargetType;
```
Comment (maxkatz6) [inline src/Markup/Avalonia.Markup.Xaml.Loader/CompilerExtensions/Transformers/AvaloniaXamlIlSelectorTransformer.cs:213]: Fix of yet one more case with compiledbinding+templatedparent

File: src/Markup/Avalonia.Markup.Xaml/MarkupExtensions/CompiledBindings/CommandAccessorPlugin.cs [hunk 3]
```diff
@@ -145,7 +145,7 @@ private void SubscribeToChanges()
             {
                 if (_dependsOnProperties is { Count:>0 } && _reference.TryGetTarget(out var o) && o is INotifyPropertyChanged inpc)
                 {
-                    WeakEventHandlerManager.Subscribe<INotifyPropertyChanged, PropertyChangedEventArgs, InpcPropertyAccessor>(
+                    WeakEventHandlerManager.Subscribe<INotifyPropertyChanged, PropertyChangedEventArgs, CommandAccessor>(
```
Comment (maxkatz6) [inline src/Markup/Avalonia.Markup.Xaml/MarkupExtensions/CompiledBindings/CommandAccessorPlugin.cs:148]: Seems to be copy paste error. CC @grokys

File: src/Avalonia.Dialogs/ManagedFileChooserFilterViewModel.cs (hunk 1 above)
Comment (kekekeks) [reply src/Avalonia.Dialogs/ManagedFileChooserFilterViewModel.cs]: I think we need to make those `RaiseAndSetIfChanged<T>` methods  to be `protected internal` then

Comment (kekekeks) [reply src/Avalonia.Dialogs/ManagedFileChooserFilterViewModel.cs]: And rename `InternalViewModelBase` to `AvaloniaDialogsInternalViewModelBase`

Comment (maxkatz6) [reply src/Avalonia.Dialogs/ManagedFileChooserFilterViewModel.cs]: Done

**Review by kekekeks — APPROVED** [review]

**Pull request merged**